### PR TITLE
Better `#to_html` for core classes

### PR DESCRIPTION
**Pull request by zverok**
Changes in the same fashion I did for `#inspect`. Mostly cosmetic, but data with multi indexes are now much more readable (for me, and may be only for me :)).

Before:

After:

(Still nothing done about multi-index dataframes using `MultiIndex` as order, though.)

### CODE REVIEW

**Comment by gnilrets** [comment]
Like!

**Comment by lokeshh** [comment]
Its amazingly great looking and much more readable from before. 

One thing worth mentioning is that when the index is not in proper order (eg `[[:a, :one], [:a, :two], [:b, :one], [:a, :three]]`) the output is a bit deceiving because in this display form one would expect all the `:a`'s and `:b`'s to lie together though this will not be the case.

File: lib/daru/index.rb [hunk 1]
```diff
@@ -386,5 +391,32 @@ def sparse_tuples
         [nil] * (cur.size - left.size) + left.map(&:first)
       }
     end
+
+    def tuples_with_rowspans
```
Comment (v0dro) [inline lib/daru/index.rb]: documentation for this method.

Comment (zverok) [inline lib/daru/index.rb]: In fact, I thought about just making it `# @private`-ly documented, it is not that easy to explain what it does and why does somebody need it :)

**Comment by zverok** [comment]
> In this case, the output will be bit deceiving because in this display form one would expect all the `:a`'s and `:b`'s to lie together though this will not be the case.

But that's how Daru stores them internally (if you'll enumerate tuples, you'll have the same result. I'm not aware of how multi-index-es are working in pandas or R, but current `#to_html` behavior just showing "the actual state of things", not trying to mask it.

**Comment by v0dro** [comment]
Yeah that's how it should be. Order needs to be preserved.

**Pull request merged**